SPECS/qemu/qemu.spec

```diff
@@ -1,64 +1,257 @@
+%global need_qemu_kvm 1
+%endif
 %ifarch x86_64
 %global kvm_package   system-x86
 # need_qemu_kvm should only ever be used by x86
 %global need_qemu_kvm 1
 %endif
+%ifarch %{power64}
+%global have_memlock_limits 1
+%global kvm_package   system-ppc
+%endif
+%ifarch s390x
+%global kvm_package   system-s390x
+%endif
+%ifarch armv7hl
+%global kvm_package   system-arm
+%endif
 %ifarch aarch64
 %global kvm_package   system-aarch64
 %endif
+%ifarch %{mips}
+%global kvm_package   system-mips
+%endif
+%ifarch riscv64
+%global kvm_package   system-riscv
+%endif
+
 %global modprobe_kvm_conf %{_sourcedir}/kvm.conf
-%ifarch x86_64
+%ifarch s390x
+    %global modprobe_kvm_conf %{_sourcedir}/kvm-s390x.conf
+%endif
+%ifarch %{ix86} x86_64
     %global modprobe_kvm_conf %{_sourcedir}/kvm-x86.conf
 %endif
+
 %global tools_only 0
+
+
+%global user_static 1
+%if 0%{?azl}
+%global user_static 0
+%endif
+%if 0%{?rhel}
+# EPEL/RHEL do not have required -static builddeps
 %global user_static 0
+%endif
+
+%global have_kvm 0
+%if 0%{?kvm_package:1}
 %global have_kvm 1
+%endif
+
+# Matches numactl ExcludeArch
 %global have_numactl 1
+%ifarch %{arm}
+%global have_numactl 0
+%endif
+
+%global have_spice 1
+%if 0%{?azl}
+%global have_spice 0
+%endif
+# Matches spice ExclusiveArch
+%ifnarch %{ix86} x86_64 %{arm} aarch64
 %global have_spice 0
+%endif
+%if 0%{?rhel} >= 9
+%global have_spice 0
+%endif
+
+# Matches xen ExclusiveArch
 %global have_xen 0
+%if 0%{?fedora}
+%ifarch x86_64 aarch64
+%global have_xen 1
+%endif
+%endif
+
+%global have_liburing 0
+%if 0%{?azl}
+%global have_liburing 1
+%endif
+%if 0%{?fedora}
+%ifnarch %{arm}
 %global have_liburing 1
+%endif
+%endif
+
+# 2.0 has this set
+%global have_virgl 0
+%if 0%{?fedora}
 %global have_virgl 1
+%endif
+
+%global have_pmem 0
+%ifarch x86_64 %{power64}
+%global have_pmem 1
+%endif
+
+%global have_jack 1
+%if 0%{?azl}
+%global have_jack 0
+%endif
+%if 0%{?rhel}
 %global have_jack 0
+%endif
+
+%global have_dbus_display 1
+%if 0%{?azl}
+%global have_dbus_display 0
+%endif
+%if %{defined rhel} && 0%{?rhel} < 9
+# RHEL/Centos 8 glib is not new enough
+%global have_dbus_display 0
+%endif
+
+%global have_libblkio 0
+%if 0%{?fedora} >= 37
+%global have_libblkio 1
+%endif
+
+%global have_gvnc_devel %{defined fedora}
+%if 0%{?azl}
+%global have_gvnc_devel 0
+%endif
+%global have_sdl_image %{defined fedora}
+%if 0%{?azl}
 %global have_sdl_image 0
+%endif
 %global have_fdt 1
 %global have_opengl 1
+%if 0%{?azl_no_ui}
+%global have_opengl 0
+%endif
 %global have_usbredir 1
 %global enable_werror 0
+
+
+# Matches edk2.spec ExclusiveArch
 %global have_edk2 0
-%global have_pmem 0
-%ifarch x86_64
-%global have_pmem 1
+%ifarch %{ix86} x86_64 %{arm} aarch64
+%global have_edk2 1
 %endif
+%if 0%{?azl}
+%global have_edk2 0
+%endif
+
 # All modules should be listed here.
 %define have_block_rbd 1
+%ifarch %{ix86} %{arm}
+%define have_block_rbd 0
+%endif
+# disable for azl till dependency is available
```

Comment:
what depenedency is this waiting on?  is this mergable without the dep?  it seems like a change from 2.0 to have this set to 0.

Reply:
Yes it is but without support of ceph-rbd. This will be deviation from 2.0. From what I know ceph-rbd is not used by baremetal as of now. Once librbd is available we can enable it back. I will file ADO work items for disabled packages so that we can revisit them as more pkgs are made available and/or we need a disabled package. Does that make sense?
My aim is to get this merged and then handle additional subpackages separately. Waiting for every subpackage to be available is very long process. Atleast if the pkg is merged we can start getting some testing done with it.

Reply:
i'm not sure i understand exactly.  if librbd existed in 2.0, it should be available as a dependency for 3.0 ... unless qemu has a min version of librdb that mariner2.0 doesn't satisfy?

Reply:
Ah, sorry for the confusion. I was waiting for the updated librbd package to be available in 3.0 (I believe that is WIP). So till that package is available, I have configured qemu to build with --disable-librbd.

Reply:
i'm still not following.  librbd exists in the 3.0 spec tree, it just isn't being built.  you should still be able to build it, right?  so you should be able to leverage that 2.0 spec code here?

Reply:
if rbd is not used by BMP, can disregard this comment from me.